code/game/objects/items/devices/mobileticketplatform.dm

```diff
@@ -0,0 +1,239 @@
+			T.source_terminal = src
+			T.date = current_date_string
+			T.time = worldtime2text()
+			D.transaction_log.Add(T)
+
+			T = new()
+			T.target_name = D.owner_name
+			T.purpose = "[ticketreason]"
+			T.amount = "[ticketamount]"
+			T.source_terminal = src
+			T.date = current_date_string
+			T.time = worldtime2text()
+			linked_account.transaction_log.Add(T)
+			print_ticket(usr)
+		else
+			to_chat(user, "<span class='warning'>Unable to process requests. User account contains insuffecient funds. Brigging is required.</span>")
+			reset()
+	else
+		to_chat(user, "<span class='warning'>Unable to access account. Check security settings and try again.</span>")
+	screen = 0
+///////////////////////////////////////////////////////TICKETS AND CHARGES/////////////////////////////////////////////////
+
+/obj/item/device/mobileticketplatform/proc/clearstringandtime()
+	ticketreason = ""//Empty the string list so we can fill it again
+	ticketamount = 0
+	ticketdescription = ""
+
+	for(C in comittedcrimes)
+		ticketreason += "[C.name]<br>"
+		ticketamount += C.max_fine
+		ticketdescription += "<small>[C.desc]</small><br>"
+
+
+/obj/item/device/mobileticketplatform/proc/print_ticket(mob/user) // Ticket Reciepts
+		var/ticketid = rand(1111,9999)
+		var/entry = {"<center><h3>Infringement Notice #[ticketid]<br></h3></center><small><b>Issued to:	</b>[ID.registered_name]<br><b>Rank:	</b>[ID.rank] <br><b>Issued at:	</b> [worldtime2text()]<br>
+					 <b>Reason for issue:	</b>[ticketreason]<br><b>Description of Space Law article:	</b>[ticketdescription]<br><b>Ticket Amount (Cr):	</b>[ticketamount]<br><b>Issuing Officer:</b>	[usr]</small>"}
+		to_chat(user, "Printing the log, standby...")
+		sleep(50)
+		playsound(loc, "sound/goonstation/machines/printer_dotmatrix.ogg", 50, 1)
+
+		var/obj/item/weapon/paper/P = new /obj/item/weapon/paper(get_turf(src))
+		P.name = "Ticket [ticketid] - [ID.registered_name] at: [worldtime2text()]"
+		P.info += {"<center>[station_name()] - Security Department</center>
+				[entry]
+				"}
+		user.put_in_hands(P)
+		ticket_logs.Add(P)
+		reset()
+
+/obj/item/device/mobileticketplatform/proc/add_charge(mob/user)
+	possible_violations.Cut()//Clear the LIST
+
+	switch(alert("Select Space Law Category.", "Space Law", "Minor", "Custom Article", "Abort"))
+		if("Abort")
+			return
+		if("Minor")
+			for(var/minorcrimes in subtypesof(/datum/spacelaw/minor))
+				possible_violations += new minorcrimes()
+		if("Custom Article")
+			var/datum/spacelaw/S = new()
+			S.name = input(user, "Please select custom article to add for [ID.registered_name]") as null|text
+			if(!isnull(S.name))
+				S.max_fine = input(user, "Please select amount to fine for [ID.registered_name]") as num
+				S.desc = S.name
+				comittedcrimes.Add(S)
+				clearstringandtime()
+
+	var/datum/spacelaw/selectedcrime = input(user, "Please select article to add for [ID.registered_name]") as null|anything in possible_violations
+	if(!isnull(selectedcrime))
```

Comment:
erm, unless I'm really misunderstanding the whole `!isnull()` is superfluous, can't you just do `if(selectedcrime)`?